### ISSUE TITLE
Consider noting that retention doesn't work with the Restrictive IAM policy

### ISSUE THREAD

**Issue by kalafut** [benbjohnson/litestream.io#36]
I'm pretty new to Litestream and copy/pasted in the [Restrictive IAM Policy](https://github.com/benbjohnson/litestream.io/blob/main/content/guides/s3/index.md#restrictive-iam-policy) to get going. After a while I noticed WALs and snapshots building up, vs. the 24h default retention I expected. I thought it might be a config error, or some operational error, but no errors were logged*. Eventually I realized the IAM user needed s3:DeleteObject in the policy.

If my experience is as expected (which I'm not sure of), I think it might be useful to add a note to the description of the restrictive policy. I'm happy to PR that.

*I was a little surprised to not see errors if the retention couldn't be met

**Comment by benbjohnson** [comment]
@kalafut Thanks for the heads up. That was a mistake on my part. I added the `s3:DeleteObject` to the policy in #39 and I added a ticket to fix the logging issue in https://github.com/benbjohnson/litestream/issues/236.

**Comment by kalafut** [comment]
I wondered if these are related...

I've been using Litestream for only a couple of weeks, and just against a tiny hobby project that has a couple dozen DB writes per day. I just got my September AWS bill and there were nearly 2M "PUT, COPY, POST, or LIST requests" (compared to the ~0 normally)! I'm just using the default settings for Litestream and highly suspect that this could have been caused by it being unable to complete the retention process. Fortunately I caught it (and it was only $9) and I'll keep a close eye on the S3 tallies through October.
